static/js/publisher/form/media.js

```diff
@@ -0,0 +1,225 @@
+    input.hidden = "hidden";
+    input.addEventListener("change", () => {
+      this.mediaChange(input);
+    });
+
+    const form = ReactDOM.findDOMNode(this).closest("form"); // eslint-disable-line react/no-find-dom-node
+    form.appendChild(input);
+    input.click();
+  }
+
+  keyboardEvent(e) {
+    if (e.key === "Enter") {
+      this.addImage();
+    }
+  }
+
+  focused(url) {
+    this.setState({
+      focused: url
+    });
+  }
+
+  blurred(url) {
+    if (this.state.focused === url) {
+      this.setState({
+        focused: null
+      });
+    }
+  }
+
+  renderBlankMedia(numberOfBlank) {
+    const blankMedia = [];
+    for (let i = 0; i < numberOfBlank; i++) {
+      const classes = mediaClasses.slice(0);
+      classes.push("is-empty");
+
+      if (i === 0) {
+        classes.push("p-listing-images__add-image");
+
+        blankMedia.push(
+          <div
+            className={classes.join(" ")}
+            key={`blank-${i}`}
+            onClick={this.addImage}
+            onKeyDown={this.keyboardEvent}
+            onFocus={this.focused.bind(this, "new")}
+            onBlur={this.blurred.bind(this, "new")}
+            tabIndex="0"
+            ref={item => (this.blankItem = item)}
+          >
+            <span role="button" className="u-align-text--center">
+              <i className="p-icon--plus" />
+              <br />
+              Add image
+            </span>
+          </div>
+        );
+      } else {
+        blankMedia.push(
+          <div className={classes.join(" ")} key={`blank-${i}`} />
+        );
+      }
+    }
+
+    return blankMedia;
+  }
+
+  render() {
+    const mediaList = this.state.mediaData.filter(
+      item => item.status !== "delete"
+    );
+
+    let blankMedia = 0;
+
+    if (mediaList.length < this.props.mediaLimit) {
+      blankMedia = this.props.mediaLimit - mediaList.length;
+    }
+
+    return (
+      <Fragment>
+        {mediaList.map((item, i) => (
+          <MediaItem
+            key={item.url}
+            url={item.url}
+            type={item.type}
+            status={item.status}
+            remove={this.markForDeletion}
+            overflow={i > 4}
+            focused={this.focused}
+            blurred={this.blurred}
+            focus={this.state.focused}
+          />
+        ))}
+        {this.renderBlankMedia(blankMedia)}
+      </Fragment>
+    );
+  }
+}
+
+Media.defaultProps = {
+  mediaLimit: 5,
+  mediaData: [],
+  updateState: () => {}
+};
+
+Media.propTypes = {
+  mediaLimit: PropTypes.number,
+  mediaData: PropTypes.array,
+  updateState: PropTypes.func
+};
+
+function initMedia(mediaHolder, images, updateState) {
+  const mediaHolderEl = document.querySelector(mediaHolder);
+  if (!mediaHolderEl) {
+    throw new Error("No media holder El");
+  }
+
+  ReactDOM.render(
+    <Media mediaData={images} updateState={updateState} />,
+    mediaHolderEl
+  );
+}
+
+export { Media as default, MediaItem, initMedia };
```

Comment:
I think it would be better to separate components to their own files and have any functionality that uses them (`initMedia`) separately.
Also, why exporting `MediaItem` here when it seems to have its own file?

Reply:
:+1: will do this (and remove the MediaItem export - legacy of when these all used to be in the same file)